app/models/instance_filter.rb

```diff
@@ -40,12 +46,33 @@ def scope_for(key, value)
     end
   end
 
+  def status_scope(value)
+    # The `join where` queries here look like they have a bug due to `domain_block`
+    # vs `domain_blocks`, however the table is `domain_blocks` while the relation on
+    # Instances is `domain_block`
+    case value.to_sym
+    when :allowed
+      Instance.joins(:domain_allow).reorder(Arel.sql('domain_allows.id desc'))
```

Comment:
Minor style thing - you can probably write these orders as `reorder(domain_allows: { id: :desc })`

(As of rails 7.2 change to support nested hash style in more AR-methods)

Same for various lines below this one